### ISSUE TITLE
Pipfile.lock hash => too many CI run

### ISSUE THREAD

**Issue by sbrunner** [dependabot/dependabot-core#3511]
If Dependabot detect 3 packages to be updated...:
- He open 3 pull request
- When we merge on the toe other are in conflict then they will be rebased
- ...

Then we will have 6 CI run, and if we have 10 packages we will have about 50 CI run!!!

Is it possible to have a solution to limit the number of CI run?

For example, by adding a parameter `max_concurent_ci_run` with we can set it to one, and we can limit the CI run it me merge the Pull request fast enough?

**Comment by brrygrdn** [comment]
Hi @sbrunner, that's an interesting suggestion - Dependabot generally treats your CI configuration as a black box, i.e. any PR it opens activates CI according to the normal rules you have in place for any contributor.

A workaround option would be to to set an `open-pull-requests-limit` value in your dependabot configuration file, see:
https://docs.github.com/en/code-security/supply-chain-security/configuration-options-for-dependency-updates#open-pull-requests-limit

Along with a daily schedule, this can be used to open a single upgrade per day so you keep upgrading gradually with less CI churn, or open 2-3 as an acceptable amount in the event one upgrade is blocked for some reason. 

There are a few things we are working on that we hopeful will improve this experience, the most significant is providing the option to group multiple updates into a single PR: https://github.com/github/roadmap/issues/148

**Comment by sbrunner** [comment]
Good to know, thanks :-)

**Comment by gberaudo** [comment]
Hi @brrygrdn, github/roadmap#148 would be a step forward, but having dependabot open 1 single PR containing all the dependency updates of the day/week would be really appreciated.

**Comment by brrygrdn** [comment]
@gberaudo Yeah that's a good call out, we intend to tackle the scenario described in github/roadmap#148 but once we set the precedent for 'rolling up' your Dependabot diffs into single PR we'll explore other configuration options such as this.

**Comment by MartinThoma** [comment]
You might be interested in https://github.com/dependabot/dependabot-core/issues/4057

**Comment by jeffwidman** [comment]
This is definitely painful, I lived this exact scenario at my previous day job before I joined the Dependabot team.

However, I'm going to close this as non-actionable, as we don't want Dependabot to cross the abstraction boundary into getting opinionated about your CI runs... too many variables for how folks run CI.

Instead, something like #4057 or https://github.com/dependabot/dependabot-core/issues/4545 would give you some tools to slow down Dependabot / delay Dependabot so to help alleviate the pain here... that or grouped updates.
